apps/judicial-system/web/src/utils/hooks/useSections/index.ts

```diff
@@ -479,54 +479,60 @@ const useSections = () => {
+      },
+      {
+        type: 'SUB_SECTION',
+        name: formatMessage(sections.indictmentsCourtSection.subpoena),
+        href: isReceptionAndAssignmentStepValid(workingCase)
+          ? `${constants.INDICTMENTS_SUBPOENA_ROUTE}/${workingCase.id}`
+          : undefined,
+      },
+      {
+        type: 'SUB_SECTION',
+        name: formatMessage(
+          sections.indictmentsCourtSection.prosecutorAndDefender,
+        ),
+        href:
+          isReceptionAndAssignmentStepValid(workingCase) &&
+          isSubpoenaStepValid(workingCase)
+            ? `${constants.INDICTMENTS_PROSECUTOR_AND_DEFENDER_ROUTE}/${workingCase.id}`
+            : undefined,
+      },
+      {
+        type: 'SUB_SECTION',
+        name: formatMessage(sections.indictmentsCourtSection.courtRecord),
+        href:
+          isReceptionAndAssignmentStepValid(workingCase) &&
+          isSubpoenaStepValid(workingCase) &&
+          isprosecutorAndDefenderStepValid(workingCase)
+            ? `${constants.INDICTMENTS_COURT_RECORD_ROUTE}/${workingCase.id}`
+            : undefined,
+      },
+    ]
 
     return {
       name: formatMessage(sections.courtSection.title),
-      children: [
-        {
-          type: 'SUB_SECTION',
-          name: formatMessage(sections.indictmentsCourtSection.overview),
-          href: `${constants.INDICTMENTS_COURT_OVERVIEW_ROUTE}/${id}`,
-        },
-        {
-          type: 'SUB_SECTION',
-          name: formatMessage(
-            sections.indictmentsCourtSection.receptionAndAssignment,
-          ),
-          href: `${constants.INDICTMENTS_RECEPTION_AND_ASSIGNMENT_ROUTE}/${workingCase.id}`,
-        },
-        {
-          type: 'SUB_SECTION',
-          name: formatMessage(sections.indictmentsCourtSection.subpoena),
-          href: isReceptionAndAssignmentStepValid(workingCase)
-            ? `${constants.INDICTMENTS_SUBPOENA_ROUTE}/${workingCase.id}`
-            : undefined,
-        },
-        {
-          type: 'SUB_SECTION',
-          name: formatMessage(
-            sections.indictmentsCourtSection.prosecutorAndDefender,
-          ),
-          href:
-            isReceptionAndAssignmentStepValid(workingCase) &&
-            isSubpoenaStepValid(workingCase)
-              ? `${constants.INDICTMENTS_PROSECUTOR_AND_DEFENDER_ROUTE}/${workingCase.id}`
-              : undefined,
-        },
-        {
-          type: 'SUB_SECTION',
-          name: formatMessage(sections.indictmentsCourtSection.courtRecord),
-          href:
-            isReceptionAndAssignmentStepValid(workingCase) &&
-            isSubpoenaStepValid(workingCase) &&
-            isprosecutorAndDefenderStepValid(workingCase)
-              ? `${constants.INDICTMENTS_COURT_RECORD_ROUTE}/${workingCase.id}`
-              : undefined,
-        },
-      ],
+      children:
+        activeSubSection === undefined
+          ? subsections.map((subsection) => {
+              return { name: '', type: '', href: subsection.href }
```

Comment:
Why is this needed? Is it to remove the subsections?

Reply:
Initially, I set this up so that this function would return an empty array if there was no activeSubsection and by doing that we could render a form stepper with an active section but no subsections, i.e. something like this (in this case the active section would be section nr2, but whatever.)

<img width="266" alt="Screenshot 2022-09-27 at 11 15 34" src="https://user-images.githubusercontent.com/3789875/192511335-c8194efe-87d2-45ca-95d5-c222312fee96.png">

There was however an issue with this approach because it made [this line](https://github.com/island-is/island.is/blob/main/apps/judicial-system/web/src/routes/Shared/Cases/Cases.tsx#L181) fail because there is no href.

This returns an array with subsections with empty names and type but includes the hrefs.

Reply:
Would it be more readable to check if subsections are empty and then define a fallback route? Like done here https://github.com/island-is/island.is/blob/473606d590a4c4799c7a9ea7bb4bc116303e7438/apps/judicial-system/web/src/routes/Shared/Cases/Cases.tsx#L196-L198

Reply:
Totally! I'll get right on that 💪